staging/src/k8s.io/apiserver/pkg/authentication/config/validation/validation.go

```diff
@@ -0,0 +1,204 @@
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, validateIssuer(authenticator.Issuer, fldPath.Child("issuer"))...)
+	allErrs = append(allErrs, validateClaimValidationRules(authenticator.ClaimValidationRules, fldPath.Child("claimValidationRules"))...)
+	allErrs = append(allErrs, validateClaimMappings(authenticator.ClaimMappings, fldPath.Child("claimMappings"))...)
+
+	return allErrs
+}
+
+func validateIssuer(issuer api.Issuer, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, validateURL(issuer.URL, fldPath.Child("url"))...)
+	allErrs = append(allErrs, validateClientIDs(issuer.ClientIDs, fldPath.Child("clientIDs"))...)
+	allErrs = append(allErrs, validateCertificateAuthority(issuer.CertificateAuthority, fldPath.Child("certificateAuthority"))...)
+
+	return allErrs
+}
+
+func validateURL(issuerURL string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(issuerURL) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath, "URL is required"))
+		return allErrs
+	}
+
+	u, err := url.Parse(issuerURL)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, issuerURL, err.Error()))
+		return allErrs
+	}
+	if u.Scheme != "https" {
+		allErrs = append(allErrs, field.Invalid(fldPath, issuerURL, "URL scheme must be https"))
+	}
+	if u.User != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, issuerURL, "URL must not contain a username or password"))
+	}
+	if len(u.RawQuery) > 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, issuerURL, "URL must not contain a query"))
+	}
+	if len(u.Fragment) > 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, issuerURL, "URL must not contain a fragment"))
+	}
+
+	return allErrs
+}
+
+func validateClientIDs(clientIDs []string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(clientIDs) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath, fmt.Sprintf(atLeastOneRequiredErrFmt, fldPath)))
+		return allErrs
+	}
+	// This stricter validation is because the --oidc-client-id flag option is singular.
+	// This will be removed when we support multiple clientIDs with the StructuredAuthenticationConfiguration feature gate.
+	if len(clientIDs) > 1 {
+		allErrs = append(allErrs, field.TooMany(fldPath, len(clientIDs), 1))
+		return allErrs
+	}
+
+	for i, clientID := range clientIDs {
+		fldPath := fldPath.Index(i)
+		if len(clientID) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath, "client ID can't be empty"))
+		}
+	}
+
+	return allErrs
+}
+
+func validateCertificateAuthority(certificateAuthority []byte, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(certificateAuthority) == 0 {
+		return allErrs
+	}
+	_, err := cert.NewPoolFromBytes(certificateAuthority)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, string(certificateAuthority), err.Error()))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Invalid(fldPath, "<omitted>", err.Error()))
```

I suspect this error message will explode with noise if you have a CA bundle with a typo...